ddtrace/pin.py

```diff
@@ -143,7 +143,7 @@ def override(
     def enabled(self):
         # type: () -> bool
         """Return true if this pin's tracer is enabled."""
-        return bool(self.tracer) and self.tracer.enabled
+        return bool(self.tracer) and (self.tracer.enabled or self.tracer._apm_opt_out)
```

Comment:
Hiding this complexity behind `tracer.enabled`, possibly by making that attribute into a property, would reduce the cognitive load incurred by future editors of code in `Pin` and elsewhere.

Reply:
Hmm... I was trying to avoid that. I'm actually setting tracer.enabled to False at some point just to be explicit about APM being disabled, not because we need to.

My point was to highlight the places where the "library" needs to be enabled regardless of the APM product being enabled.

I'd rather remove that `tracer.enabled = False`, if you prefer